packages/doenetml-worker/lib-doenetml-core/src/dast/extended_vec.rs

```diff
@@ -0,0 +1,92 @@
+use std::{
+    ops::{Index, IndexMut},
+    slice::Iter,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ExtendedVector<T> {
+    inner: Vec<T>,
+    index_offset: isize,
+}
+
+impl<T> ExtendedVector<T> {
+    pub fn new(index_offset: isize) -> ExtendedVector<T> {
+        ExtendedVector {
+            inner: Vec::new(),
+            index_offset,
+        }
+    }
+
+    pub fn from_vector(values: Vec<T>, index_offset: isize) -> ExtendedVector<T> {
+        ExtendedVector {
+            inner: values,
+            index_offset,
+        }
+    }
+
+    pub fn len(&self) -> usize {
```

Comment:
We need a warning docstring at least. `v[v.len() - 1]` may panic, which is not the usual behaviour of a vector.